### PR TITLE
Add libnegf package

### PR DESCRIPTION
**Pull request by awvwgk**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by awvwgk** [comment]
@conda-forge/staged-recipes This recipe for a MPI-aware Fortran library is ready for review. Targets are Linux and OSX.

File: recipes/libnegf/build.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+   "-DWITH_MPI=${MPI}"
+   "-GNinja"
+   ".."
+)
+
+mkdir -p _build
+pushd _build
+cmake "${cmake_options[@]}"
+
+ninja all install
```
Comment (ocefpaf) [inline recipes/libnegf/build.sh:23]: Is there a make check/test or ctest that we can run here?

File: recipes/libnegf/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  run:
+    - {{ mpi }}  # [mpi != "nompi"]
+    - {{ pin_compatible('mpifx') }} mpi_{{ mpi }}_*  # [mpi != 'nompi']
+    - libblas
+    - liblapack
+
+test:
+  commands:
+    - test -f $PREFIX/lib/libnegf.dylib  # [osx]
+    - test -f $PREFIX/lib/libnegf.so  # [linux]
```
Comment (ocefpaf) [inline recipes/libnegf/meta.yaml:56]: Please [use `SHLIB_EXT`](https://github.com/conda-forge/suitesparse-feedstock/blob/a3425e96b4ca6aecb6bb957dc4f0042ca7052c9c/recipe/meta.yaml#L38) and make this a single line.

File: recipes/libnegf/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+about:
+  home: https://github.com/{{ name }}/{{ name }}
```
Comment (ocefpaf) [inline recipes/libnegf/meta.yaml:59]: The jinja proliferation like this is not really readable or nice. One cannot click on that URL and open the documentation when reading the recipe.

File: recipes/libnegf/build.sh (hunk 1 above)
Comment (awvwgk) [reply recipes/libnegf/build.sh:23]: The only testsuite for libnegf I'm aware of is available in a dependent project.

**Comment by ocefpaf** [comment]
CIs did not start :-/ Let's close and open it to kick them.

**Pull request merged**